application/core.py

```diff
@@ -0,0 +1,236 @@
+    hs = window.winfo_screenheight()
+    x = (ws / 2) - (w / 2)
+    y = (hs / 2) - (h / 2)
+    if main_menu:
+        return '{}x{}+{}+{}'.format(w, h, int(x), int(y))
+    if custom_wh:
+        return '{}x{}+{}+{}'.format(custom_wh[0], custom_wh[1], int(x), int(y))
+    return '+{}+{}'.format(int(x), int(y))
+
+
+class Application(tk.Tk):
+    """Base of the application."""
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        # Initial window features
+        self.geometry(get_window_info(self, main_menu=True))
+        self.iconbitmap('a2r.ico')
+
+        # Wrapper for frames
+        container = tk.Frame(self)
+        container.pack()
+        container.grid_rowconfigure(0, weight=1)
+        container.grid_columnconfigure(0, weight=1)
+        self.frames = {}
+        for F in (MainMenu,):
+            page_name = F.__name__
+            frame = F(parent=container, controller=self)
+            self.frames[page_name] = frame
+            frame.grid(row=0, column=0, sticky="nsew")
+
+        self.show_frame("MainMenu")
+        self.title("async2rewrite")
+
+    def show_frame(self, page_name):
+        """Show a frame for the given page name."""
+        frame = self.frames[page_name]
+        frame.tkraise()
+
+
+class MainMenu(tk.Frame):
+    """Main menu of the application."""
+    def __init__(self, parent, controller):
+        super().__init__(parent)
+
+        # Basic declarations
+        self.parent = parent
+        self.controller = controller
+
+        # Label
+        a2r_img = tk.PhotoImage(file='../logo.png')
+        label = tk.Label(self, image=a2r_img)
+        label.image = a2r_img
+        label.grid(row=1, column=1, sticky="we")
+
+        # Menu buttons
+
+        snippet = tk.Button(self, text="Convert Snippet", command=lambda: Snippet(self.parent))
+        snippet.grid(row=2, column=1, sticky="we")
+
+        file = tk.Button(self, text="Convert File", command=self.file)
+        file.grid(row=3, column=1, sticky="we")
+
+        notices = tk.Button(self, text="Credits", command=lambda: Notices(self.parent))
+        notices.grid(row=4, column=1, sticky="we")
+
+        # Configure grid geometry
+        self.grid_rowconfigure(1, weight=1)
+        self.grid_columnconfigure(1, weight=1)
+
+    def file(self):
+        """Converts a file from async to rewrite."""
+        name = filedialog.askopenfilename(initialdir=".", title="Select a file",
+                                          filetypes=(("Python Files", "*.py"),))  # Asks for original file
+        save = filedialog.asksaveasfilename(initialdir=".", title="Save file as",
+                                            filetypes=(("Python Files", "*.py"),))  # Asks where to save the file
+
+        try:
+            file_result = async2rewrite.from_file(name)  # Converts code from original file to rewrite
+            with open(save, 'w') as f:
+                f.write(file_result.replace('\t', '    '))

```

Comment:
This does not work. `async2rewrite.from_file` returns a dictionary, not a string.